systemd/systemd.go

```diff
@@ -352,6 +369,11 @@ type Error struct {
 	exitCode int
 }
 
+// NewError creates a systemd.Error, exported for mocks during testing
+func NewError(exitCode int, cmd []string, msg []byte) *Error {
```

Comment:
If it's just for tests this should be in export_test.go. There's no reason right now for external logic to create these errors.

Reply:
I guess I don't follow, we can't put this function inside the `wrappers/export_test.go` because we need to access private members of the systemd.Error struct and the `wrappers/export_test.go` package is not `systemd,` it's `wrappers`.The only place during testing we could put this function would be in `systemd/export_test.go`, which doesn't allow us to access this error while running the `wrappers` tests, which is the actual place we need this.

@bboozzoo thoughts on this problem? I think putting it in the systemd package as you suggested initially is the simplest solution here.

Reply:
@niemeyer the problem here is that `systemctl is-enabled <foo>`, exits with error and prints `disabled` when the service is indeed disabled. To mock that nicely we need to generate `systemd.Error{exitCode:1, msg: []byte("disabled"}` in a different package.

I suppose we could rename this to `MockError()` to make it more explicit that this is only for the tests and restore the original code here https://github.com/snapcore/snapd/pull/5777/files#diff-42c07ec7ffa970857b4e0db2614e482bR56 Would that work for you?